pkg/istio/rbac/v2/provider_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2021, Verizon Media Inc.
+// Licensed under the terms of the 3-Clause BSD license. See LICENSE file in
+// github.com/yahoo/k8s-athenz-istio-auth for terms.
+package v2
+
+import (
+	"github.com/ardielle/ardielle-go/rdl"
+	"github.com/stretchr/testify/assert"
+	"github.com/yahoo/athenz/clients/go/zms"
+	"github.com/yahoo/k8s-athenz-istio-auth/pkg/athenz"
+	fakev1 "github.com/yahoo/k8s-athenz-syncer/pkg/client/clientset/versioned/fake"
+	adInformer "github.com/yahoo/k8s-athenz-syncer/pkg/client/informers/externalversions/athenz/v1"
+	"istio.io/api/security/v1beta1"
+	workloadv1beta1 "istio.io/api/type/v1beta1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config/schema/collections"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"testing"
+	"time"
+)
+
+const (
+	DomainName = "test.namespace"
+	username   = "user.name"
+	username1  = "user.*"
+)
+
+var (
+	onboardedService = &k8sv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "onboarded-service",
+			Namespace: "test-namespace",
+			Annotations: map[string]string{
+				"authz.istio.io/enabled": "true",
+			},
+			Labels: map[string]string{
+				"app": "productpage",
+			},
+		},
+	}
+)
+
+func TestConvertAthenzModelIntoIstioRbac(t *testing.T) {
```

Comment:
Can we add more unit tests here to cover the same cases we're covering for the provider v1 flow in a subsequent PR?

Reply:
will do, having a pending pr for it https://github.com/yahoo/k8s-athenz-istio-auth/pull/48